rrmngmnt/foreman_host_wrapper.py

```diff
@@ -0,0 +1,186 @@
+from host import Host
+from foreman.client import Foreman
+
+
+HOSTS_COLLECTION = "hosts"
+
+# Host status constants
+FOREMAN_HOST_STATUS_NO_CHANGES = "No changes"
+FOREMAN_HOST_STATUS_OUT_OF_SYNC = "Out of sync"
+FOREMAN_HOST_STATUS_PENDING_INSTALLATION = "Pending Installation"
+
+
+class ForemanException(Exception):
+    pass
+
+
+class ForemanHost(Host):
+    """
+    Foreman wrapper class
+    """
+
+    def __init__(
+        self, host_ip, foreman_url, foreman_user,
+        foreman_password, api_version=2
+    ):
+        """
+        Initialize ForemanApi class
+
+        :param host_ip: host ip
+        :type host_ip: str
+        :param foreman_url: foreman url
+        :type foreman_url: str
+        :param foreman_user: foreman user
+        :type foreman_user: str
+        :param foreman_password: foreman password
+        :type foreman_password: str
+        :param api_version: foreman api version
+        :type api_version: int
+        """
+        super(ForemanHost, self).__init__(host_ip)
+        self.foreman_api = Foreman(
+            url=foreman_url,
+            auth=(
+                foreman_user, foreman_password
+            ),
+            api_version=api_version
+        )
+        self.host_id = self.get_element_id(self.fqdn, HOSTS_COLLECTION)
+
+    def get_element_id(
+        self, element_name, collection_name, search_attr="name"
+    ):
+        """
+        Get element id from foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :param collection_name: foreman collection name
+        :type collection_name: str
+        :param search_attr: search by attr
+        :type search_attr: str
+        :return: element id
+        :rtype: str
+        """
+        self.logger.info(
+            "Looking for %s under foreman collection %s",
+            element_name, collection_name
+        )
+        collection = getattr(self.foreman_api, collection_name)
```

Comment:
I use this function to get id also for other collections, but I agreed it not good design at all.
In general it must be something like:

class ForemanCollection(object):
    collection_name = None
    def __init__(self, jenkins_url, jenkins_user, jenkins_password, api_version=2):
         self._f = ...
         self.collection = getattr(self._f, self.collection_name)
    
    def get_element_id(self, element_name, search_attr="name"):
        ....
    def add(self, kwargs):
         ....
    def update(self, kwargs):
         ....
    def delete(self):
        ....

class ForemanHostCollection(ForemanCollection):
    collection_name = "hosts"

but I do not sure how I can integrate together Host class and ForemanHostCollection class